### PR TITLE
Template pytorch docker file

### PR DESCRIPTION
**Pull request by verdimrc**
*Issue #, if available:* N/A

*Description of changes:* provide a template Dockerfile to create a PyTorch docker image. The Dockerfile contains optional stanzas, and expect users to read through it, understand what each stanza does, then create their own Dockerfile. This is amalgamation of a wide range of images, notably those targeted for p4, p5, and latest Parallel Cluster.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by perifaws** [comment]
@verdimrc, we'll need to talk more about this one with the team as it's changing the naming convention. Should we move the AMI directory name to AMI + containers?

**Comment by verdimrc** [comment]
For sure, let's talk about it. For now, I move the Dockerfile to sit together with the AMI templates.

**Comment by perifaws** [comment]
Recommended changes:
```
2.AMI_and_containers
|- 1.AmazonMachineImage
|- 2.DockerContainer
```

**Comment by verdimrc** [comment]
looks good. I'll reorganize that subdir.

**Comment by verdimrc** [comment]
@perifaws : push latest commits. PR ready for re-review.

- re-org to dir. structure stated in https://github.com/aws-samples/awsome-distributed-training/pull/37#issuecomment-1810425239
- add readme
- bugfixes

File: 2.ami_and_containers/2.docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+# Compute nodes (aka build nodes) are transient, so we need to keep the docker image on shared fs,
+# which head node can load into its local registry.
+#
+#     # Build node: save image to file
+#     docker save nvidia-pt-od:2310 > /fsx/nvidia-pt-od__2310.tar
+#
+#     # Load image to local docker registry -> on head node, or new compute/build node.
+#     docker load < /fsx/nvidia-pt-od__2310.tar
+####################################################################################################
+FROM nvcr.io/nvidia/pytorch:23.10-py3
```
Comment (perifaws) [inline 2.ami_and_containers/2.docker/Dockerfile:22]: It's pulling from NVCR, what do you think of renaming this like `0.nvcr-pytorch.aws-optimized.Dockerfile` or equivalent? So we have container files for multiple sources (pytorch, aws). 


@sean-smith thoughts on that?

Comment (sean-smith) [reply 2.ami_and_containers/2.docker/Dockerfile:22]: Do we have another container image source?

Comment (verdimrc) [reply 2.ami_and_containers/2.docker/Dockerfile]: @perifaws: renamed.

@sean-smith: nemo frameworks. Also possibilities of other base PyTorch image.

**Review by perifaws — APPROVED** [review]

File: 2.ami_and_containers/2.docker/Dockerfile (hunk 1 above)
Comment (perifaws) [reply 2.ami_and_containers/2.docker/Dockerfile]: Let's finish & resolve this conversation before merging. @sean-smith anything more on this PR?

Comment (sean-smith) [reply 2.ami_and_containers/2.docker/Dockerfile]: nope

**Comment by mhuguesaws** [comment]
Please address conflicts.

**Pull request merged**